pkg/apis/core/validation/validation_test.go

```diff
@@ -26715,3 +26715,59 @@ func TestValidatePodResize(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateNodeSwapStatus(t *testing.T) {
+	var nilSwapCapacity *int64 = nil
+	makeNode := func(swapCapacity *int64) core.Node {
+		node := makeNode("test-node", nil)
+		if swapCapacity != nil {
+			node.Status.NodeInfo.Swap = &core.NodeSwapStatus{
```

Comment:
What about when `node.Status.NodeInfo.Swap == nil` ?

Reply:
Perhaps you can reframe it as

```
makeNode := func(swap *NodeSwapStatus) *Node { ... }
makeSwapStatus := func(capacity int64) *NodeSwapStatus { ... }
```

and then the testcases are:

```
makeNode(nil) // nil swap status
makeNode(&NodeSwapStatus{}) // nil capacity
makeNode(makeSwapStatus(123456))
makeNode(makeSwapStatus(0))
makeNode(makeSwapStatus(-123456))
```

Reply:
@thockin Done. PTAL.